src/libraries/System.Private.CoreLib/src/System/Threading/Volatile.cs

```diff
@@ -194,40 +220,44 @@ public static ulong Read(ref readonly ulong location) =>
         [NonVersionable]
         public static void Write(ref ulong location, ulong value) =>
             Write(ref Unsafe.As<ulong, long>(ref location), (long)value);
```

Comment:
This could be simplified to the same pattern as the others, removing the unsafe code.